notebook/static/notebook/js/notebook.js

```diff
@@ -164,29 +164,79 @@ import {ShortcutEditor} from 'notebook/js/shortcuteditor';
         var that = this;
 
         Object.defineProperty(this, 'line_numbers', {
-          get: function(){
-            var d = that.config.data||{}
-            var cmc =  (d['Cell']||{})['cm_config']||{}
-            return cmc['lineNumbers'] || false;
-          },
-          set: function(value){
-            this.get_cells().map(function(c) {
-              c.code_mirror.setOption('lineNumbers', value);
-            })
-            that.config.update({'Cell':{'cm_config':{'lineNumbers':value}}})
-          }
-          
-        })
+            get: function() {
```

Comment:
I tried to update the line numbers toggle logic to use `ConfigWithDefaults` (via `that.class_config.get_sync` and `that.class_config.set`) to be consistent with header and toolbar but it looks like the line numbers toggle status is being read from `notebook.config.data` regardless. `notebook.json` looks like:

```json
{
  "Notebook": {
    "Toolbar": false,
    "Cell": {
      "cm_config": {
        "lineNumbers": false
      }
    },
    "Header": false
  },
  "Cell": {
    "cm_config": {
      "lineNumbers": true
    }
  }
}
```

The top-level `Cell` is set using `notebook.config.update` while `Notebook.Cell` is set using `notebook.class_config.set`. Changing the value of `Notebook.Cell.cm_config.lineNumbers` has no effect even after changing the line numbers toggle logic, so the cells/Codemirror appear to be accessing `Cell.cm_config.lineNumbers` directly vs. accessing `notebook.lineNumbers`. I know this is probably confusing and that's because it is! It doesn't really matter because everything is working but it would be nice if these 3 properties shared the same implementation.